pygmt/tests/test_meca.py

```diff
@@ -0,0 +1,282 @@
+"""
+Tests for meca
+"""
+import os
+import pandas as pd
+import numpy as np
+import pytest
+
+from .. import Figure
+
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+
+
+@pytest.mark.mpl_image_compare
+def test_meca_spec_dictionary():
+    """
+    Test supplying a dictionary containing a single focal mechanism to the
+    `spec` argument.
+    """
+
+    fig = Figure()
+
+    # Right lateral strike slip
+    fig.meca(
+        dict(strike=0, dip=90, rake=0, magnitude=5),
+        longitude=0,
+        latitude=5,
+        depth=0,
+        scale="2.5c",
+        region=[-1, 4, 0, 6],
+        projection="M14c",
+        frame=2,
+    )
+
+    # Left lateral strike slip
+    fig.meca(
+        dict(strike=0, dip=90, rake=180, magnitude=5),
+        longitude=2,
+        latitude=5,
+        depth=0,
+        scale="2.5c",
+    )
+
+    # Thrust
+    fig.meca(
+        dict(strike=0, dip=45, rake=90, magnitude=5),
+        longitude=0,
+        latitude=3,
+        depth=0,
+        scale="2.5c",
+    )
+    fig.meca(
+        dict(strike=45, dip=45, rake=90, magnitude=5),
+        longitude=2,
+        latitude=3,
+        depth=0,
+        scale="2.5c",
+    )
+
+    # Normal
+    fig.meca(
+        dict(strike=0, dip=45, rake=-90, magnitude=5),
+        longitude=0,
+        latitude=1,
+        depth=0,
+        scale="2.5c",
+    )
+    fig.meca(
+        dict(strike=45, dip=45, rake=-90, magnitude=5),
+        longitude=2,
+        latitude=1,
+        depth=0,
+        scale="2.5c",
+    )
+
+    # Mixed
+    fig.meca(
+        dict(strike=10, dip=35, rake=129, magnitude=5),
+        longitude=3.4,
+        latitude=0.6,
+        depth=0,
+        scale="2.5c",
+    )
+
+    return fig
```

Comment:
Why plot so many beachballs in this test?

Reply:
Sorry to be too picky here. We're trying our best not to add big images into the repository, as it will significantly increase the whole repository size. 

All other tests looks good to me. This test figure is as large as ~50 KB, but most beachballs seem unnecessary to me.

Reply:
Whoops, this is another leftover from when I was just wrapping the GMT tests. Agreed.

Reply:
Addressed in 50f582c